src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1311,6 +1315,86 @@ private function registerPropertyInfoConfiguration(array $config, ContainerBuild
         }
     }
 
+    /**
+     * Loads lock configuration.
+     *
+     * @param array            $config
+     * @param ContainerBuilder $container
+     * @param XmlFileLoader    $loader
+     */
+    private function registerLockConfiguration(array $config, ContainerBuilder $container, XmlFileLoader $loader)
+    {
+        $loader->load('lock.xml');
+
+        $container->getDefinition('lock.store.flock')->replaceArgument(0, sys_get_temp_dir());
+
+        // configure connectable stores
+        foreach (array('redis', 'memcached') as $store) {
+            if ($this->isConfigEnabled($container, $config[$store]) && count($config[$store]['hosts']) > 0) {
+                /** @var Reference[] $hostsDefinitions */
+                $hostsDefinitions = array();
+                foreach ($config[$store]['hosts'] as $host) {
+                    $definition = new ChildDefinition('lock.store.'.$store.'.abstract');
+
+                    // generate a service connection for the host
+                    $container->resolveEnvPlaceholders($host, null, $usedEnvs);
+                    if ($usedEnvs || preg_match('#^[a-z]++://#', $host)) {
+                        $dsn = $host;
+
+                        if (!$container->hasDefinition($host = 'lock.connection.'.$store.'.'.md5($dsn))) {
+                            $connectionDefinition = new Definition(\stdClass::class);
+                            $connectionDefinition->setPublic(false);
+                            $connectionDefinition->setFactory(array($container->getDefinition($definition->getParent())->getClass(), 'createConnection'));
+                            $connectionDefinition->setArguments(array($dsn));
+                            $container->setDefinition($host, $connectionDefinition);
+                        }
+                    }
+
+                    $definition->replaceArgument(0, new Reference($host));
+                    $container->setDefinition($name = 'lock.store.'.$store.'.'.md5($host), $definition);
+
+                    $hostsDefinitions[] = new Reference($name);
+                }
+
+                if (count($hostsDefinitions) > 1) {
+                    $definition = new ChildDefinition('lock.store.combined.abstract');
+                    $definition->replaceArgument(0, $hostsDefinitions);
```

Comment:
Would it make sense to lazy load stores using an iterator? It seems the combined store can break in the middle of the store chain. Btw, why not `ChainStore` instead of `CombinedStore` for consistency?